plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateCommandWrapper.java

```diff
@@ -132,8 +132,9 @@ Use VIR_DOMAIN_XML_SECURE (value = 1) prior to v1.0.0.
             vmsnapshots = libvirtComputingResource.cleanVMSnapshotMetadata(dm);
 
             Map<String, MigrateCommand.MigrateDiskInfo> mapMigrateStorage = command.getMigrateStorage();
+            boolean migrateStorage = MapUtils.isNotEmpty(mapMigrateStorage);
```

Comment:
Would it not be safer to make this boolean a final one? That way you are 100% sure it's not changed anywhere.

Reply:
@wido Sounds good. I can do that and then put the code on the branch that @rhtyd asked.